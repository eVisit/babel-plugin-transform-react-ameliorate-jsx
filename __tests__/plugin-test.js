const babel = require("@babel/core"),
      FS = require('fs'),
      PATH = require('path'),
      pluginReactAmeliorateJSX = require('../src');

// Date mock
require('jest-mock-now')();

function transform(code, extraOpts, extraBabelOpts) {
  return new Promise((resolve, reject) => {
    var babelOpts = Object.assign({
      code: true,
      ast: false,
      babelrc: false,
      plugins: [
        [pluginReactAmeliorateJSX, Object.assign({}, extraOpts || {})]
      ]
    }, extraBabelOpts);

    babel.transform(
      ('' + code),
      babelOpts,
      function(error, result) {
        if (error)
          return reject(error);

        resolve(result);
      }
    );
  });
}

function transformFile(fileName, extraOpts) {
  var fullFileName = (PATH.join(__dirname, 'test-code', fileName) + '.js');
  return transform(FS.readFileSync(fullFileName), extraOpts, { filename: fullFileName });
}

describe('babel-preset-react-ameliorate', function () {
  it('should work with default options', async function() {
    var result = await transformFile('test01');
    expect(result.code).toMatchSnapshot();
  });

  it('should work with custom options', async function() {
    var result = await transformFile('test02', {
      pragma: '_customCreateElement,React.createElement',
      pragmaFrag: '_customCreateFragmentElement,React.Fragment',
    });

    expect(result.code.indexOf('_customCreateElement ||')).toBeGreaterThan(-1);
    expect(result.code.indexOf('_customCreateFragmentElement ||')).toBeGreaterThan(-1);
    expect(result.code).toMatchSnapshot();
  });

  it('should work with inline options', async function() {
    var result = await transformFile('test03');
    expect(result.code.indexOf('dom1 || dom2')).toBeGreaterThan(-1);
    expect(result.code.indexOf('DomFrag1 || DomFrag2')).toBeGreaterThan(-1);
    expect(result.code).toMatchSnapshot();
  });
});
