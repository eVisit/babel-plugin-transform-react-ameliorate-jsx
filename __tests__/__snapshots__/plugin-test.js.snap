// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-react-ameliorate should work with custom options 1`] = `
"function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';

class MyTestComponent extends React.Component {
  render() {
    return (_customCreateElement || React.createElement)(_customCreateFragmentElement || React.Fragment, null, (_customCreateElement || React.createElement)(View, _extends({
      hello: true
    }, opts), (_customCreateElement || React.createElement)(React.Fragment, null, (_customCreateElement || React.createElement)(Text, {
      stuff: true
    }, \\"Hello world!\\"), (_customCreateElement || React.createElement)(\\"div\\", null, (_customCreateElement || React.createElement)(\\"span\\", null, \\"Cool beans\\")))));
  }

}"
`;

exports[`babel-preset-react-ameliorate should work with default options 1`] = `
"function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';

class MyTestComponent extends React.Component {
  render() {
    return (this && this._raCreateElement || React.createElement)(React.Fragment, null, (this && this._raCreateElement || React.createElement)(View, _extends({
      hello: true
    }, opts), (this && this._raCreateElement || React.createElement)(React.Fragment, null, (this && this._raCreateElement || React.createElement)(Text, {
      stuff: true
    }, \\"Hello world!\\"), (this && this._raCreateElement || React.createElement)(\\"div\\", null, (this && this._raCreateElement || React.createElement)(\\"span\\", null, \\"Cool beans\\")))));
  }

}"
`;

exports[`babel-preset-react-ameliorate should work with inline options 1`] = `
"function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

/** @jsx dom1,dom2 */

/** @jsxFrag DomFrag1,DomFrag2 */
import React from 'react';

class MyTestComponent extends React.Component {
  render() {
    return (dom1 || dom2)(DomFrag1 || DomFrag2, null, (dom1 || dom2)(View, _extends({
      hello: true
    }, opts), (dom1 || dom2)(React.Fragment, null, (dom1 || dom2)(Text, {
      stuff: true
    }, \\"Hello world!\\"), (dom1 || dom2)(\\"div\\", null, (dom1 || dom2)(\\"span\\", null, \\"Cool beans\\")))));
  }

}"
`;
